src/core/lib/resource_quota/memory_quota.cc

```diff
@@ -487,6 +487,78 @@ BasicMemoryQuota::InstantaneousPressureAndMaxRecommendedAllocationSize() {
 
 namespace memory_quota_detail {
 
+double PressureController::Update(double error) {
+  bool is_low = error < 0;
+  bool was_low = absl::exchange(last_was_low_, is_low);
+  double new_control;  // leave unset to compiler can note bad branches
+  if (is_low && was_low) {
+    // Memory pressure is too low this round, and was last round too.
+    // If we have reached the min reporting value last time, then we will report
+    // the same value again this time and can start to increase the ticks_same_
+    // counter.
+    if (last_control_ == min_) {
+      ticks_same_++;
+      if (ticks_same_ >= max_ticks_same_) {
+        // If it's been the same for too long, reduce the min reported value
+        // down towards zero.
+        min_ /= 2.0;
+        ticks_same_ = 0;
+      }
+    }
+    // Target the min reporting value.
+    new_control = min_;
+  } else if (!is_low && !was_low) {
+    // Memory pressure is high, and was high previously.
+    ticks_same_++;
+    if (ticks_same_ >= 100) {
```

Comment:
Could we change this to ticks_same_ >= max_ticks_same_ ?

Reply:
or maybe it should be more aggressive when memory pressure is high. perhaps max_ticks_same_/2 ?

Reply:
done